### PR TITLE
[FAB-17145] Add test network to Fabric Samples

### PR DESCRIPTION
**Pull request by nikhil550**
Addresses: FabricFAB-17147
New sample is meant to be long term replacement for BYFN. 
- New sample is simpler and more organized
- Bypasses tools container for org1 and org2
- Provides more options than the e2e flow of BYFN
- Provides support for using CA's in addition to cryptogen

The network is meant to be run on Fabric 2.0. To test the sample, you need to build the latest images from master locally. You also need to build the latest binaries for cryptogen and peer, and move then into the bin folder of your local fabric samples. You also need to copy the latest configuration files into the config file inside fabric samples. These are done when you clone the fabric samples, images, and binaries.

Signed-off-by: NIKHIL E GUPTA <negupta@us.ibm.com>

### CODE REVIEW

File: test-network/network.sh [hunk 1]
```diff
@@ -0,0 +1,565 @@
...
+      echo "Failed to generate certificates..."
+      exit 1
+    fi
+
+  fi
+
+  # Create crypto material using Fabric CAs
+  if [ "$CRYPTO" == "Certificate Authorities" ]; then
+
+    fabric-ca-client version > /dev/null 2>&1
```
Comment (nikhil550) [inline test-network/network.sh]: @mbwhite Added this section to download the fabric-ca-client if it is not available locally.

**Comment by nikhil550** [comment]
The script assumes that you followed the steps to install the sample and the binaries: https://hyperledger-fabric.readthedocs.io/en/latest/install.html

This downloads the binaries in a bin folder and the latest configuration files in a config folder. You can copy the binaries and configuration files to those locations to test.

File: test-network/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+## Running the test network
+
+Use the ./network.sh script to stand up a simple Fabric test network with two
+peer organizations with one peer each and a single node raft ordering service.
+You can also use the script to create channels, and deploy the fabcar chaincode
+on those channels.
+
+Before you can deploy the test network, you need follow the instructions to 
+[Install the Samples, Binaries and Docker Images](https://hyperledger-fabric.readthedocs.io/en/latest/install.html) in the Hyperledger Fabric documentation.
```
Comment (nikhil550) [inline test-network/README.md]: Let folks now they need to install the binaries.

File: test-network/network.sh [hunk 3]
```diff
@@ -0,0 +1,573 @@
...
+BLACKLISTED_VERSIONS="^1\.0\. ^1\.1\.0-preview ^1\.1\.0-alpha"
+
+# Do some basic sanity checking to make sure that the appropriate versions of fabric
+# binaries/images are available. In the future, additional checking for the presence
+# of go or other items could be added.
+function checkPrereqs() {
+  ## Check if your have cloned the peer binaries and configuration files.
+  peer version > /dev/null 2>&1
+
+  if [[ $? -ne 0 || ! -d "../config" ]]; then
```
Comment (nikhil550) [inline test-network/network.sh]: Let folks know and exit if the config files and binaries are missing.

**Comment by nikhil550** [comment]
> FYI - going to pull this locally and give it a test run :-)
> 
>     * network.sh - would be good to add `set -euo pipefail` at the top so any failures, eg missing commands stop the script immediately
> 
>     * I can't see if there's a 'clear config' command.  So for example it's not possible to force regeneration of the crypto material.
> 
>     * fabric-ca-client missing - doesn't give a warning
> 
>     * core.yaml missing?

@mbwhite on the points above
- Seems to encounter problems running `set -euo pipefail`
- I remove the certs every time I run /network.up. Leaving certs around could cause errors with left around artifacts. These artifacts are also cleared when you run ./network down
- Added command download fabric-ca-client if not found
- You need to download core.yaml with the binaries. Added checks to see if they are there. See above.

**Comment by rameshthoomu** [comment]
Good work @nikhil550 
1) This script will fail when you build images and binaries. Config directory is missing. 2019-12-09 17:41:25.710 EST [main] InitCmd -> ERRO 001 Fatal error when initializing core config : FABRIC_CFG_PATH /Users/thoomu@us.ibm.com/gopath/src/github.com/hyperledger/fabric-samples/test-network/../config/ does not exist

**Comment by nikhil550** [comment]
> Good work @nikhil550
> 
>     1. This script will fail when you build images and binaries.
>        Config directory is missing. 2019-12-09 17:41:25.710 EST [main] InitCmd -> ERRO 001 Fatal error when initializing core config : FABRIC_CFG_PATH /Users/thoomu@us.ibm.com/gopath/src/github.com/hyperledger/fabric-samples/test-network/../config/ does not exist

Thanks @rameshthoomu 
Addressed with most recent changes above.

**Comment by nikhil550** [comment]
Added some changes in response to feedback from @rameshthoomu 
- Added the generate back to the add Org3 script and cleaned up the help text
- Fixed the path in the org3 script to point to the right binaries
- Added retry to the createChannel command to allow raft to elect leader 
- redid the joinchannel with retry to match other commands in the script

File: test-network/README.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
+## Running the test network
+
+Use the `./network.sh` script to stand up a simple Fabric test network. The
+network has two peer peer organizations with one peer each and a single node
+raft ordering service. You can also use the script to create channels, and deploy
+the fabcar chaincode on those channels. The test network is being introduced in
```
Comment (nikhil550) [inline test-network/README.md]: Per @lehors suggestion, added a sentence that this will be a long term replacement for BYFN.

**Comment by lehors** [comment]
I'm going to ignore the CI error which is unrelated to this change and merge it.

**Pull request merged**